### PR TITLE
KAFKA-14133: Migrate storeMetadata mock in StoreChangelogReaderTest to Mockito

### PR DESCRIPTION
**Pull request by clolov**
This pull request takes a similar approach to how TaskManagerTest is being migrated to Mockito mock by mock for easier reviews.

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 1]
```diff
@@ -226,8 +225,7 @@ public void shouldSupportUnregisterChangelogBeforeInitialization() {
         setupStateManagerMock();
         final Map<TaskId, Task> mockTasks = mock(Map.class);
         EasyMock.expect(mockTasks.get(null)).andReturn(mock(Task.class)).anyTimes();
-        EasyMock.expect(storeMetadata.offset()).andReturn(9L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:229]: Reported as unused by Mockito

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 2]
```diff
@@ -257,15 +255,16 @@ public void shouldSupportUnregisterChangelogBeforeInitialization() {
...
         setupStateManagerMock();
+        setupStoreMetadata();
         final Map<TaskId, Task> mockTasks = mock(Map.class);
         EasyMock.expect(mockTasks.get(null)).andReturn(mock(Task.class)).anyTimes();
         EasyMock.expect(mockTasks.containsKey(null)).andReturn(true).anyTimes();
-        EasyMock.expect(storeMetadata.offset()).andReturn(9L).anyTimes();
-        EasyMock.expect(storeMetadata.endOffset()).andReturn(10L).anyTimes();
+        when(storeMetadata.offset()).thenReturn(9L);
         if (type == STANDBY) {
+            when(storeMetadata.endOffset()).thenReturn(10L);
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:264]: Only used on the STANDBY path

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 3]
```diff
@@ -363,12 +363,12 @@ public void shouldSupportUnregisterChangelogAfterCompletion() {
     @Test
     public void shouldInitializeChangelogAndCheckForCompletion() {
         setupStateManagerMock();
+        setupStoreMetadata();
         final Map<TaskId, Task> mockTasks = mock(Map.class);
         EasyMock.expect(mockTasks.get(null)).andReturn(mock(Task.class)).anyTimes();
         EasyMock.expect(mockTasks.containsKey(null)).andReturn(true).anyTimes();
-        EasyMock.expect(storeMetadata.offset()).andReturn(9L).anyTimes();
-        EasyMock.expect(storeMetadata.endOffset()).andReturn(10L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:370]: Unused according to Mockito

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 4]
```diff
@@ -453,17 +456,17 @@ private void shouldPollWithRightTimeout(final boolean stateUpdaterEnabled) {
...
+        setupStoreMetadata();
         final Properties properties = new Properties();
         shouldPollWithRightTimeout(properties);
     }
 
     private void shouldPollWithRightTimeout(final Properties properties) {
         final TaskId taskId = new TaskId(0, 0);
 
-        EasyMock.expect(storeMetadata.offset()).andReturn(null).andReturn(9L).anyTimes();
-        EasyMock.expect(storeMetadata.endOffset()).andReturn(10L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:464]: Ditto

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 5]
```diff
@@ -821,11 +831,11 @@ public Map<TopicPartition, OffsetAndMetadata> committed(final Set<TopicPartition
     @Test
     public void shouldThrowIfEndOffsetsFail() {
         setupActiveStateManager();
+        when(storeMetadata.changelogPartition()).thenReturn(tp);
         final TaskId taskId = new TaskId(0, 0);
 
-        EasyMock.expect(storeMetadata.offset()).andReturn(10L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:826]: Ditto

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 6]
```diff
@@ -862,10 +873,9 @@ public void shouldRequestCommittedOffsetsAndHandleTimeoutException() {
         EasyMock.expectLastCall();
 
         when(stateManager.changelogAsSource(tp)).thenReturn(true);
-        EasyMock.expect(storeMetadata.offset()).andReturn(5L).anyTimes();
-        EasyMock.expect(storeMetadata.endOffset()).andReturn(10L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:866]: Ditto

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java [hunk 7]
```diff
@@ -924,13 +934,13 @@ public synchronized ListConsumerGroupOffsetsResult listConsumerGroupOffsets(fina
     @Test
     public void shouldThrowIfCommittedOffsetsFail() {
         setupStateManagerMock();
+        when(storeMetadata.changelogPartition()).thenReturn(tp);
 
         final TaskId taskId = new TaskId(0, 0);
 
         when(stateManager.taskId()).thenReturn(taskId);
         when(stateManager.changelogAsSource(tp)).thenReturn(true);
-        EasyMock.expect(storeMetadata.offset()).andReturn(10L).anyTimes();
```
Comment (clolov) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java:932]: Ditto

**Comment by divijvaidya** [comment]
@clolov we have test failures related to this PR. Can you please fix them: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15116/1/testReport/org.apache.kafka.streams.processor.internals/StreamThreadTest/Build___JDK_11_and_Scala_2_13___shouldNotEnforceRebalanceWhenCurrentlyRebalancing_0_/

**Comment by divijvaidya** [comment]
My bad! the test failure I notes above is known to be flaky as per https://ge.apache.org/scans/tests?search.relativeStartTime=P28D&search.rootProjectNames=kafka&search.tags=trunk&search.timeZoneId=Europe%2FBerlin&tests.container=org.apache.kafka.streams.processor.internals.StreamThreadTest&tests.test=shouldNotEnforceRebalanceWhenCurrentlyRebalancing%5B0%5D and is not related to this PR.

**Review by divijvaidya — APPROVED** [review]

**Comment by lucasbru** [comment]
@clolov I do believe that the flaky test @divijvaidya mentioned started with another mockito migration PR on 4th of december, #13932. Maybe you could give it a look even if it's not related to this PR.

**Comment by clolov** [comment]
Okay @lucasbru, I will look into it

**Pull request merged**